modules/xmpp/ChatRoom.js

```diff
@@ -510,10 +513,14 @@ ChatRoom.prototype.onPresenceUnavailable = function (pres, from) {
             delete this.members[i];
             this.onParticipantLeft(i, member.isFocus);
         }
+        this.connection.emuc.doLeave(this.roomjid);
+        this.eventEmitter.emit(XMPPEvents.MUC_LEFT);
```

Comment:
Yes you are right, the presence when kicked out is having both <status code='307'/> and <status code='110'/> so we will fire both. 
Will reorder those if statements.